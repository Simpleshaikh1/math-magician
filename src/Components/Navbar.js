import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { Link, useLocation } from 'react-router-dom';

const links = [
  { path: '/', text: 'Home' },
  { path: '/calculator', text: 'Calculator' },
  { path: '/quotes', text: 'Quotes' },
];

const Navbar = () => {
  const location = useLocation();

  const isActive = (path) => (location.pathname === path ? { color: 'purple' } : {});

  return (
    <nav>
      <ul>
        {links.map((link) => (
          <li key={link.text}>
            <Link to={link.path} style={isActive(link.path)}>
              {link.text}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
