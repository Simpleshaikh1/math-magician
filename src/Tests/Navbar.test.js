import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../Components/Navbar';

test('renders Navbar correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});
