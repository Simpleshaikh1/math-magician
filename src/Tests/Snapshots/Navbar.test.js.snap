// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="/"
          style="color: orange;"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</div>
`;